extensions/rules_haskell_dependencies.bzl

```diff
@@ -23,9 +24,27 @@ def repositories(*, bzlmod):
 
     # TODO: Remove when tests are run with a ghc version containing Cabal >= 3.10
     # See https://github.com/tweag/rules_haskell/issues/1871
-    http_archive(
-        name = "Cabal",
-        build_file_content = """
+    if GHC_VERSION and GHC_VERSION.startswith("9.4."):
+        http_archive(
+            name = "Cabal",
```

Comment:
Where is `@Cabal` all used? From what I can tell it's mostly used for test dependencies and for the worker. So, it shouldn't directly affect users of rules_haskell, correct?

Side note, I noticed that some test-dependencies in rules_haskell's `MODULE.bazel`, e.g. some `stack_snapshot`s, are not marked as dev-dependencies. It's out of scope for this PR, but something we should fix.

Reply:
> Where is `@Cabal` all used? From what I can tell it's mostly used for test dependencies and for the worker. So, it shouldn't directly affect users of rules_haskell, correct?

Yes.

> Side note, I noticed that some test-dependencies in rules_haskell's `MODULE.bazel`, e.g. some `stack_snapshot`s, are not marked as dev-dependencies. It's out of scope for this PR, but something we should fix.

Agreed, and I already tried that (setting `dev_dependency = True`) but that failed with some wild error later, since rules_haskell_tests depended on it.